.github/workflows/publish.yml

```diff
@@ -4,12 +4,32 @@ on:
   push:
     branches: # Snapshots
       - main
+      - gradle # TODO: remove this line
     tags: ["**"] # Releases
   release:
     types: [published]
 
 jobs:
-  publish-artifacts:
+  publish-snapshot-artifacts:
     name: Publish / Artifacts
+    if: github.ref_type == 'branch' # Snapshots
     uses: playframework/.github/.github/workflows/publish.yml@v3
+    with:
+      gradle-build-root: gradle-twirl
+      cmd: |
+        sbt ci-release
+        cd gradle-twirl
+        ./gradlew --no-daemon publishToSonatype -x test -PsonatypeUsername="$SONATYPE_USERNAME" -PsonatypePassword="$SONATYPE_PASSWORD"
+    secrets: inherit
+
+  publish-release-artifacts:
+    name: Publish / Artifacts
+    if: github.ref_type == 'tag' # Releases
+    uses: playframework/.github/.github/workflows/publish.yml@v3
+    with:
+      gradle-build-root: gradle-twirl
+      cmd: |
+        sbt ci-release
+        cd gradle-twirl
+        ./gradlew --no-daemon publishPlugins -x test -Pgradle.publish.key="$GRADLE_PUBLISH_KEY" -Pgradle.publish.secret="$GRADLE_PUBLISH_SECRET"
```

Comment:
We will see on next non-snapshot release if everything works :wink: (but everything looks fine so far)

Reply:
Don't forget to add `GRADLE_PUBLISH_KEY` and `GRADLE_PUBLISH_SECRET` in organization secrets :wink:
Without these release will be failed in 💯 percent 😄